From 524f4b9641a75563896b7aff3d69759334039b13 Mon Sep 17 00:00:00 2001
From: Sowmiya Sree Elavalagan <quic_ssreeela@quicinc.com>
Date: Tue, 5 Nov 2024 16:01:58 +0530
Subject: [PATCH] Kconfig: Add a new Kconfig QCOM_RPROC_DISABLE_MPD_SUPPORT

Add new Kconfig which would enable DT flag to build the
necessary DTS property.

Change-Id: I71c3df0e939bd58948df3f4240255511fa92ebb8
Signed-off-by: Sowmiya Sree Elavalagan <quic_ssreeela@quicinc.com>
---
 drivers/remoteproc/Kconfig | 6 ++++++
 scripts/Makefile.lib       | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/drivers/remoteproc/Kconfig b/drivers/remoteproc/Kconfig
index a03e5abe8e7c..eeac77182297 100644
--- a/drivers/remoteproc/Kconfig
+++ b/drivers/remoteproc/Kconfig
@@ -305,6 +305,12 @@ config QCOM_Q6V5_WCSS_SEC
 	  for the Hexagon V5 based WCSS remote processors on e.g. IPQ54xx.
 	  This is a TME-L based secure wireless subsystem.
 
+config QCOM_RPROC_DISABLE_MPD_SUPPORT
+	bool "Disable MPD support"
+	depends on QCOM_Q6V5_WCSS_SEC
+	help
+	   Disable MPD driver support and enable UserPD for AHB device from client driver
+
 config QCOM_SYSMON
 	tristate "Qualcomm sysmon driver"
 	depends on RPMSG
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 4bca0258e77d..610e3d1899cb 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -257,6 +257,10 @@ else ifeq ($(CONFIG_IPQ_MEM_PROFILE),512)
 dtc_cpp_flags += -D __IPQ_MEM_PROFILE_512_MB__
 endif
 
+ifeq ($(CONFIG_QCOM_RPROC_DISABLE_MPD_SUPPORT),y)
+dtc_cpp_flags += -D __RPROC_DISABLE_MPD_SUPPORT__
+endif
+
 ifeq ($(CONFIG_CPU_THERMAL),y)
 dtc_cpp_flags += -D __CPU_THERMAL__
 endif
-- 
2.34.1

